include/rcutils/get_env.h

```diff
@@ -47,9 +47,8 @@ extern "C"
  * printf("Valued of 'SOME_ENV_VAR': %s\n", env_value);
  * ```
  *
- * Environment variables will be truncated at 2048 characters on Windows.
- *
- * This function is not thread-safe.
+ * This function cannot be thread-safely called together with rcutils_set_env
+ * (or any platform specific equivalent), but multiple calls to this function are thread safe.
```

Comment:
```suggestion
 * This function cannot be concurrently called together with rcutils_set_env (or any platform specific equivalent)
 * on different threads, but multiple concurrent calls to this function are thread safe.
```

Also, while we are in here, can you please update the documentation for `rcutils_get_home_dir`?  It has the same constraints as this call (also, it no longer truncates to 2048 characters).

Reply:
> Also, while we are in here, can you please update the documentation for rcutils_get_home_dir? It has the same constraints as this call (also, it no longer truncates to 2048 characters).

Ah I didn't see that one, I will fix it.

Reply:
Fixed in https://github.com/ros2/rcutils/pull/291/commits/967e833daa8f62803b42fca01498e8484b464088.

I also corrected a bit `set_env` docs in https://github.com/ros2/rcutils/pull/291/commits/56ce9ef02e859d4ad30789e20b9aefdbc9296620.
`set_env` isn't thread safe, either if you're using `get_env` or not.